src/compiler/parser.ts

```diff
@@ -3008,7 +3009,9 @@ module ts {
             node.parameters.pos = parameter.pos;
             node.parameters.end = parameter.end;
 
-            parseExpected(SyntaxKind.EqualsGreaterThanToken);
+            if ((node.arrow = parseExpectedToken(SyntaxKind.EqualsGreaterThanToken, false, Diagnostics._0_expected, "=>"))) {
+                node.arrow.parent = node;

```

Comment:
You shouldn't set any parent explicitly while parsing.  It will happen during binding automatically.